jupyter_core/application.py

```diff
@@ -115,14 +117,19 @@ def config_files(self):
         if self.config_file:
             return [self.config_file]
         else:
-            return []
+            return [self.config_file_name]
```

Comment:
Also, if I'm following the logic in traitlets right, even when we have a full file path in `.config_file`, it will be loaded again for each entry in `.config_file_paths`. In most cases I don't think that will be a problem, but it's not very elegant.